src/kili/presentation/client/tag.py

```diff
@@ -64,12 +65,16 @@ def tag_project(
         if tag_ids is None:
             if tags is None:
                 raise ValueError("Either `tags` or `tag_ids` must be provided.")
-            tag_ids = tag_use_cases.get_tag_ids_from_labels(labels=tags)
+            tag_ids = tag_use_cases.get_tag_ids_from_labels(
```

Comment:
this should be part of the `tag_project` use case or if it will be used in several useCases. It is rather a gateway operation. a useCase = one client action

Reply:
The useCase can take a tags argument that is a dataclass that either have a tag_id field or a tag_label field.
We have a similar issue with assets if we want to switch from a "list of attributes" approach to a "list of dict with attributes" one: how to assert that an asset can be well identified if its id or external ids are optional but that at least one should be provided ?
I began to think about it but didn't do any implem because we didn't move asset mutations yet. 
It can maybe done with Pydantic validators ?

Reply:
> this should be part of the `tag_project` use case or if it will be used in several useCases. It is rather a gateway operation. a useCase = one client action

I asked Marc and it's fine to call multiple use cases sequentially in the presenter. 

Making a big use case makes things harder to read/maintain/test IMO

If you want to put that in a gateway operation, you're going to end up with the gateway calling itself (self.list_tags_by_org), which is bad. A layer method should not call another layer method (from what I understand of clean archi...)